content/en/blog/_posts/2023-08-21-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: blog
+title: "Kubernetes 1.28: New Job Features"
+date: 2023-08-21
+slug: kubernetes-1-28-jobapi-update
+---
+
+**Authors:** Kevin Hannon (G-Research), Michał Woźniak (Google)
+
+This blog discusses two new features in Kubernetes 1.28 to improve Jobs for batch
+users: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+and [BackoffLimitPerIndex](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs).
+
+## Pod Replacement Policy
+
+### What problem does this solve?
+
+By default, when a pod enters a terminating state (e.g. due to preemption or
+eviction), a replacement pod is created immediately, and both pods are running
+at the same time.
+
+This is problematic for some popular machine learning frameworks, such as
+TensorFlow and [JAX](https://jax.readthedocs.io/en/latest/), which require at most one pod running at the same time,
+for a given index (see more details in the [issue](https://github.com/kubernetes/kubernetes/issues/115844)).
+
+Creating the replacement Pod before the previous one fully terminates can also
+cause problems in clusters with scarce resources or with tight budgets. These
+resources can be difficult to obtain so pods can take a long time to find
+resources and they may only be able to find nodes until the existing pods are
+fully terminated. Further, if cluster autoscaler is enabled, the replacement
+Pods might produce undesired scale ups.
+
+### How can I use it
+
+This is an alpha feature, which you can enable by turning on `JobPodReplacementPolicy`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in
+your cluster.
+
+Once the feature is enabled you can use it by creating a new Job, which specifies
+`podReplacementPolicy` field as shown here:
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
```

Comment:
explain how this is interpreted: The pods would only be replaced once they reached the Failed phase, and not when they are terminating.

Reply:
Added